packages/@ama-sdk/core/src/fwk/api.helpers.ts

```diff
@@ -115,3 +116,53 @@ export function tokenizeRequestOptions(tokenizedUrl: string, queryParameters: {
 
   return {values, url: tokenizedUrl, queryParams: tokenizedQueryParams};
 }
+
+/**
+ * Compute the reviver to use in case the success response code returned by the API does not match the API
+ * Fallback to the lowest status code's reviver.
+ * Does not try to match non-successful responses as error are handled separately
+ *
+ * @param revivers
+ * @param endpoint
+ * @param response
+ * @param disableFallback
+ */
+export function getResponseReviver<T>(revivers: {[statusCode: number]: ReviverType<T> | undefined} | undefined | ReviverType<T>, response: {status?: number; ok?: boolean} | undefined,
+  endpoint?: string, options?: {disableFallback?: boolean}): ReviverType<T> | undefined {
+  const logPrefix = `API status code error for ${endpoint || 'unknown'} endpoint`;
+  if (!response?.ok) {
```

Comment:
Why `?` as the `response` is mandatory anyway.

Moreover, the fact that `ok` is optional is leading to issue:
if `ok === undefined` it does not mean that the response is "not ok", it just means that the `ok` property is not provided

So there are 2 possible fixes:

- `ok` is mandatory:

```typescript
if (!response.ok) {
  return undefined;
}
```

- `ok` is optional:

```typescript
if (response.ok === false || (!response.ok && !!response.status && (response.status < 200 || response.status > 299))) {
  return undefined;
}
```